packages/appstore/src/components/modals/real-wallets-upgrade/components/wallet-link/wallet-link-wrapper.tsx

```diff
@@ -16,58 +18,53 @@ export type TWalletLinkWrapper = {
     }[];
 };
 
-const WalletLinkWrapper = observer(({ wallet_details, account_list }: TWalletLinkWrapper) => {
+const WalletLinkWrapper = ({ wallet }: { wallet: TWalletLinkWrapper }) => {
```

Comment:
Lets avoid inline typing like this. Separate type declaration for better code readability. If more props are added, it will be difficult to type. Just a suggestion

Reply:
Agree! And it looks like you have wallet with type like this wallet: { wallet_details: ..., account_list: {}}. So you are wrapping all props inside of the wallet, don't understand why.
```suggestion
const WalletLinkWrapper = ({ wallet_details, account_list }: TWalletLinkWrapper) => {
```

Reply:
Hi @thisyahlen-deriv,  I have done this to avoid destructuring of the data passed from WalletLinkingStep to WalletLinkWrapper for each wallet. If you think that I should destructure the data in WalletLinkingStep then I can change it 🤔

Reply:
Yes man you can destructure it. Plus i think walletStep doesnt need to have data passed as prop. We can just call the mocked data or the in future the API in the component itself which is `WalletLinkingStep`.

Reply:
Hey @thisyahlen-deriv,
Ok, I have destructred the prop passed to `WalletLinkWrapper` but I am still passing the data as props from `WalletSteps` as the data is split over there based on the pages (data[0] = Non-EU page, data[1] = EU page and data[2] = Crypto page).
We may have the hook (for fetching and transforming the data) in the `WalletSteps` and render the steps as required.
I hope my explanation makes sense 😅.